python-client/giskard/models/base/wrapper.py

```diff
@@ -0,0 +1,291 @@
+            raise ValueError(
+                f"The output of your model has shape {raw_predictions.shape}, but we expect it to be (n_entries, n_classes), \n"
+                f"where `n_classes` is the number of classes in your model output ({len(self.meta.classification_labels)} in this case)."
+            )
+
+        return raw_predictions
+
+    @abstractmethod
+    def model_predict(self, data):
+        """Performs the model inference/forward pass.
+
+        Parameters
+        ----------
+        data : Any
+            The input data for making predictions. If you did not specify a
+            `data_preprocessing_function`, this will be a :class:`pd.DataFrame`,
+            otherwise it will be whatever the `data_preprocessing_function`
+            returns.
+
+        Returns
+        -------
+        numpy.ndarray
+            If the model is ``classification``, it should return an array of
+            probabilities of shape ``(num_entries, num_classes)``.
+            If the model is ``regression`` or ``text_generation``, it should
+            return an array of ``num_entries`` predictions.
+        """
+        ...
+
+    def save(self, local_path: Union[str, Path]) -> None:
+        super().save(local_path)
+        self.save_wrapper_meta(local_path)
+        if self.data_preprocessing_function:
+            self.save_data_preprocessing_function(local_path)
+        if self.model_postprocessing_function:
+            self.save_model_postprocessing_function(local_path)
+
+    @abstractmethod
+    def save_model(self, path: Union[str, Path]) -> None:
+        """Saves the wrapped ``model`` object.
+
+        Parameters
+        ----------
+        path : Union[str, Path]
+            Path to which the model should be saved.
+        """
+        ...
+
+    def save_data_preprocessing_function(self, local_path: Union[str, Path]):
+        with open(Path(local_path) / "giskard-data-preprocessing-function.pkl", "wb") as f:
+            cloudpickle.dump(self.data_preprocessing_function, f, protocol=pickle.DEFAULT_PROTOCOL)
+
+    def save_model_postprocessing_function(self, local_path: Union[str, Path]):
+        with open(Path(local_path) / "giskard-model-postprocessing-function.pkl", "wb") as f:
+            cloudpickle.dump(self.model_postprocessing_function, f, protocol=pickle.DEFAULT_PROTOCOL)
+
+    def save_wrapper_meta(self, local_path):
+        with open(Path(local_path) / "giskard-model-wrapper-meta.yaml", "w") as f:
+            yaml.dump(
+                {
+                    "batch_size": self.batch_size,
+                },
+                f,
+                default_flow_style=False,
+            )
+
+    @classmethod
+    def load(cls, local_dir, **kwargs):
+        kwargs["data_preprocessing_function"] = cls.load_data_preprocessing_function(local_dir)
+        kwargs["model_postprocessing_function"] = cls.load_model_postprocessing_function(local_dir)
+        kwargs.update(cls.load_wrapper_meta(local_dir))
```

Comment:
It's going to do the opposite of what we discussed:

`save and load (like we already do for pytorch) but prefer the values from kwargs if they're provided`

meaning that you won't be able to override the value of batch_size from the kwargs

Reply:
That’s true, I will add a test